package.json

```diff
@@ -53,8 +53,27 @@
     "@typescript-eslint/eslint-plugin": "^3.10.1",
     "@typescript-eslint/parser": "^3.10.1",
     "eslint": "^7.8.1",
+    "husky": "^4.3.0",
+    "lint-staged": "^10.3.0",
     "nodemon": "^2.0.4",
     "prettier": "^2.1.1",
     "typescript": "^4.0.2"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,json,md,yml}": [
+      "npx eslint --fix",
+      "prettier --no-config --write"
+    ],
+    "*.ts": [
+      "npx eslint --fix",
+      "prettier --no-config --write",
+      "npx tsc --esModuleInterop -m CommonJS --moduleResolution node --outDir ../dist",
```

Comment:
i noticed you tried the flag `-p tsconfig.json`, i believe it's actually `--p tsconfig.json` (2 dashes), can you give that another try?

Reply:
https://www.typescriptlang.org/docs/handbook/compiler-options.html
actually wait, maybe i'm wrong here

Reply:
I believe it would still give an error because it is on the command line. I think, like how I have to set the `outDir` to `../dist` I have to set it to `../tsconfig.json`. I will try it out tho.

Reply:
I changed it, but I don't know how to test it to see if it works.

Reply:
1. i think you were right with the flag being `-p` my bad.
2. maybe add a bogus console.log to the utility script and see if the precommit catches that and builds?

Reply:
`error TS5042: Option 'project' cannot be mixed with source files on a command line.`
Guess we have to manually type all the compiler options.

Reply:
https://github.com/microsoft/TypeScript/issues/27379
interesting rabbit hole in regards to this exact issue